rpc/src/v1/impls/eth.rs

```diff
@@ -545,6 +546,51 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Box::new(future::done(res))
 	}
 
+	fn proof(&self, address: RpcH160, values: Vec<RpcH256>, num: Trailing<BlockNumber>) -> BoxFuture<EthAccount> {
+		let a: H160 = address.clone().into();
+		let key1 = keccak(a);
+
+		let num = num.unwrap_or_default();
+		let id = match num {
+			BlockNumber::Num(n) => BlockId::Number(n),
+			BlockNumber::Earliest => BlockId::Earliest,
+			BlockNumber::Latest => BlockId::Latest,
+			BlockNumber::Pending => {
+				warn!("`Pending` is deprecated and may be removed in future versions. Falling back to `Latest`");
+				BlockId::Latest
+			}
+		};
+
+		try_bf!(check_known(&*self.client, num.clone()));
+		let res = match self.client.prove_account(key1, id) {
+			Some((proof,account)) => Ok(EthAccount {
+				address : address.into(),
+				balance : account.balance.into(),
+				nonce : account.nonce.into(),
+				code_hash : account.code_hash.into(),
+				storage_hash : account.storage_root.into(),
+				account_proof: Some( proof.iter().map(|b| Bytes::new(b.clone())).collect::<Vec<Bytes>>()),
+				storage_proof: Some( values.iter().filter_map(|storage_index| { 
```

Comment:
take ownership and skip cloning

```suggestion
				storage_proof: Some(values.into_iter().filter_map(|storage_index| { 
```

Reply:
fixed